server/src/main/java/org/elasticsearch/action/admin/indices/template/post/TransportSimulateIndexTemplateAction.java

```diff
@@ -0,0 +1,176 @@
+                simulateTemplateToAdd, request.getIndexTemplateRequest().indexTemplate());
+        }
+
+        String matchingTemplate = findV2Template(simulateOnClusterState.metadata(), request.getIndexName(), false);
+        if (matchingTemplate == null) {
+            listener.onResponse(new SimulateIndexTemplateResponse(null, null));
+            return;
+        }
+        Settings settings = resolveSettings(simulateOnClusterState.metadata(), matchingTemplate);
+
+        // empty request mapping as the user can't specify any explicit mappings via the simulate api
+        Map<String, Object> mappings = resolveV2Mappings("{}", simulateOnClusterState, matchingTemplate, xContentRegistry);
+        String mappingsJson = Strings.toString(XContentFactory.jsonBuilder()
+            .startObject()
+            .field(MapperService.SINGLE_MAPPING_NAME, mappings)
+            .endObject());
+
+        List<Map<String, AliasMetadata>> resolvedAliases = MetadataIndexTemplateService.resolveAliases(simulateOnClusterState.metadata(),
+            matchingTemplate);
+
+        // create the index with dummy settings in the cluster state so we can parse and validate the aliases
+        Settings dummySettings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(settings)
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
+            .build();
+        final IndexMetadata indexMetadata = IndexMetadata.builder(request.getIndexName()).settings(dummySettings).build();
+
+        simulateOnClusterState = ClusterState.builder(simulateOnClusterState)
+            .metadata(Metadata.builder(simulateOnClusterState.metadata())
+                .put(indexMetadata, true)
+                .build())
+            .build();
+
+        IndexService tempIndexService = indicesService.createIndex(indexMetadata, Collections.emptyList(), false);
+        final Index index = tempIndexService.index();
+        try (Closeable dummy = () -> tempIndexService.close("temp", false)) {
+            List<AliasMetadata> aliases = MetadataCreateIndexService.resolveAndValidateAliases(request.getIndexName(), Set.of(),
+                resolvedAliases, simulateOnClusterState.metadata(), aliasValidator, xContentRegistry,
+                // the context is only used for validation so it's fine to pass fake values for the
+                // shard id and the current timestamp
+                tempIndexService.newQueryShardContext(0, null, () -> 0L, null));
+
+            IndexTemplateV2 templateV2 = simulateOnClusterState.metadata().templatesV2().get(matchingTemplate);
+            assert templateV2 != null : "the matched template must exist";
+
+            Map<String, List<String>> overlapping = findConflictingV1Templates(simulateOnClusterState, matchingTemplate,
```

Comment:
I think this should put the results in a new map, otherwise we rely on `findConflictingV1Templates` returning a mutable `Map`

Reply:
ah good shout, I had allocations on my mind